### DISCUSSION TITLE
EddieHub Events/Titles

### DISCUSSION THREAD

**Discussion by ghost** [EddieHubCommunity/support#2224] [Ideas]
## Events

We can surround competitions to be spread across the year, 4 in total

- Open Source Clone (1-2 weeks): Make a clone of your favorite closed-source app
- No code Contribs (1 Day): Make no code contributions to the community
- Open API (2-3 Weeks): Make an open-source API/dataset of something, like the PokeApi
- Cloud Coding (1 Day): Code something in the cloud for the cloud

## Titles

We can also have some really small titles like:

- Contributor of the month
- No code Athlete of the month
- Most Helpful Member of the month
- Community Shoutout of the quarter

## Additional Context

Most of the titles are to be backed by [EddieHubCommunity/api](https://github.com/EddieHubCommunity/api)

## Suggestions

Some contributions aren't directly measured:
 - Sara working on the website
 - People coming on live streams

For live streams, I think we can use the YouTube API for measuring the comments.

For people working on no-code/low-code solutions, if they are not open-source/can't be open source: we shall use Community Shoutouts and Helpful Members

cc:
  - @eddiejaoude @Vyvy-vi @mikeysan @nhcarrigan @schmelto (Part of the conversations)
  - @qwe123coder (Started the conversation)
  - @Cahllagerfeld (For help with the the API stuff)

**Comment by Vyvy-vi** [comment]
The 1st one - 
> Open Source Clone (1-2 weeks): Make a clone of your favourite closed-source app

could be a great idea, however, I think we might need to specify some rules and regulations, to ensure that there isn't any copyright or trademark violations.

The 2nd one -
> No code Contribs (1 Day): Make no code contributions to the community

This could be neat, but in practice, I'm not sure if we could share access for the Katra Page builders.. 
Maybe we can make a jekyll site and docs for this?

The 3rd one -
 > Open API (2-3 Weeks): Make an open-source API/dataset of something, like the PokeApi
 
This would be really cool! We can also maybe pair this with Eddie's API workshop, and give people like a week to make great APIs
I <3 this idea.

The 4th one -
> Cloud Coding (1 Day): Code something in the cloud for the cloud

This could be great! :)

**Reply by ghost** [reply]
> could be a great idea, however, I think we might need to specify some rules and regulations, to ensure that there isn't any copyright or trademark violations.

Not sure why would this result in violations, several clones of facebook and netflix are made daily... We ain't saying them to redesign their favourite closed-source app, we are telling them to make an open source alternative for their favourite closed source app

> This could be neat, but in practice, I'm not sure if we could share access for the Katra Page builders..
Maybe we can make a jekyll site and docs for this?

We just want to encourage no-code open source contributions 😅 The main purpose is to bring no-code to open source... Similar to the no-code for open source talk in EddieCon... We just want them to contribute no-codely... Like contribution to markdown-only repositories like the awesome family or resource repositories.

> give people like a week to make great APIs

I gave them 2-3 Weeks since gathering a lot of data for the api/dataset in a week will be enormous... Rather we should give them time, to make actual projects and have something cool for their resumes instead of just MVPs

> This could be great! :)

I just realised that the description I gave for the cloud coding point was not descriptive... I meant that we will code in the cloud using tools like gitpod and codespaces for coding something that would be useful for the cloud

**Reply by Vyvy-vi** [reply]
For no-code contributions, maybe we can also run workshops/streams talking about tools like Jekyll(with gh-pages), docsify, mkdocs or Asciidoc?

I think this can also add a lot of value 🤔

**Reply by Vyvy-vi** [reply]
By trademark violations, I meant logos, trademarked design assets, etc. from those applications. (I've seen quite a few people that use the original companies art assets and logos in their clone, which I think can be a legal issue. Would be great if we can share this in the event details)

**Reply by ghost** [reply]
> By trademark violations, I meant logos, trademarked design assets, etc. from those applications. (I've seen quite a few people that use the original companies art assets and logos in their clone, which I think can be a legal issue. Would be great if we can share this in the event details)

We could specify that `We have to make alternatives not clones`. What do you think?

**Reply by ghost** [reply]
> For no-code contributions, maybe we can also run workshops/streams talking about tools like Jekyll(with gh-pages), docsify, mkdocs or Asciidoc?
> 
> I think this can also add a lot of value 🤔

Yeah that would work too 👍🏻 Not sure about jekyll but docsify, mkdocs and asciidoc would work

**Reply by qwel-exe** [reply]
oooo I did not see this . my idea started a big conversation

**Reply by Vyvy-vi** [reply]
> Yeah that would work too 👍🏻 Not sure about jekyll but docsify, mkdocs and asciidoc would work


I think Jekyll is a good no-code tool that many people might see value in using.

**Reply by ghost** [reply]
> > Yeah that would work too 👍🏻 Not sure about jekyll but docsify, mkdocs and asciidoc would work
> 
> I think Jekyll is a good no-code tool that many people might see value in using.

I meant that people could make any no-code contributions. But for the stream/workshop I suggest to use something simpler like docsify
